src/services/turret/function.ts

```diff
@@ -0,0 +1,14 @@
+import Turret from "@/entities/Turret";
+
+export async function getTxFunction(
+  turret: Turret,
+  txFunctionHash: string
+): Promise<unknown> {
+  const response = await fetch(turret.url + "/tx-functions/" + txFunctionHash);
+
+  if (!response.ok) {
+    throw new Error(await response.json());
```

Comment:
avoid doing `response.json()` twice, await for the result once and reuse it inside and outside the if